### PR TITLE
Fix #7906: Make ConfigurationSetting Equality an object reference.

### LINKED ISSUE #7906: Make ConfigurationSetting Equality an object reference.
Will make this something we'll no longer do:
https://github.com/Azure/azure-sdk-for-net/issues/5475

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

**Review by annelo-msft — APPROVED** [review]

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ namespace Azure.Data.AppConfiguration
     /// <summary>
     /// A setting, defined by a unique combination of a key and label.
     /// </summary>
-    public sealed class ConfigurationSetting : IEquatable<ConfigurationSetting>
+    public sealed class ConfigurationSetting
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs]: Why make this change, though? Looking at the bug, I don't see why this helps. `EqualityComparer<T>.Default` would use this implementation of `IEquatable<T>` anyway. This doesn't seem to add any value, and is inconsistent with what we do with "extensible enums".

Comment (annelo-msft) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs]: @heaths, could you say more about what you do with extensible enums?

Comment (heaths) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs]: We just implement `IEquatable<T>`. This way, the default comparer that many methods use just use that implementation, and you don't have to pass in a comparer to get the right behavior, i.e. "It Just Works". 🙂

Comment (annelo-msft) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs]: Per offline discussion, we will not implement IEquatable for ConfigurationSetting.  We will look into providing verbosely named Comparers after GA.

Comment (AlexanderSher) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs]: So, should I merge this PR?

Comment (annelo-msft) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs]: Yep!

**Pull request merged**